sirfilesystem.c

```diff
@@ -158,42 +158,43 @@ char* _sir_getcwd(void) {
 }
 
 char* _sir_getappfilename(void) {
-    char* buffer = (char*)calloc(SIR_MAXPATH, sizeof(char));
-    if (NULL == buffer) {
+#if defined(__linux__) || defined(__NetBSD__) || defined(__SOLARIS__) || \
+    defined(__DragonFly__) || defined(__CYGWIN__)
+# define __READLINK_OS__
+# if defined(__linux__) || defined(__CYGWIN__)
+#  define PROC_SELF "/proc/self/exe"
+# elif defined(__NetBSD__)
+#  define PROC_SELF "/proc/curproc/exe"
+# elif defined(__DragonFly__)
+#  define PROC_SELF "/proc/curproc/file"
+# elif defined(__SOLARIS__)
+#  define PROC_SELF "/proc/self/path/a.out"
+# endif
+    struct stat st;
+    if (-1 == lstat(PROC_SELF, &st)) {
         _sir_handleerr(errno);
         return NULL;
     }
 
+    size_t size = (st.st_size > 0) ? st.st_size + 1 : SIR_MAXPATH;
```

Comment:
Looks like you need an exception for 0 too:

Like the example code in https://github.com/aremmell/libsir/issues/44#issuecomment-1625511800 -
```c
           /* Some magic symlinks under (for example) /proc and /sys
              report 'st_size' as zero. In that case, take PATH_MAX as
              a "good enough" estimate. */

           if (sb.st_size == 0)
               bufsiz = PATH_MAX;
```